app/controllers/recipes_controller.rb

```diff
@@ -1,6 +1,13 @@
 class RecipesController < ApplicationController
+
   def index
-  end
+    if ingredient_id = params["ingredient"]
+      ingredient_id.to_i
```

Comment:
Because it rails. Magic everywhere. Should not works. But rails known you will need number as a id and change it for you.